src/ai/script_ai.cpp

```diff
@@ -582,11 +596,21 @@ static int CclAiDebugPlayer(lua_State *l)
 }
 
 /**
+** <b>Description</b>
+**
 **  Need a unit.
 **
 **  @param l  Lua state.
 **
 **  @return   Number of return values
+**
+** Example:
+**
+** <div class="example">
+** 	<code>
+**		unit_needed = AiNeed("unit-human-barracks")
```

Comment:
Could you make the example a bit longer? First I was confused as I read this as "the function returns the new unit". But it doesn't, right? It just returns if the unit is needed or it's already existing?

Reply:
> It just returns if the unit is needed or it's already existing?
Yes but I think I should put the example inside a function where it's executed by the A.I.
Something like this:
`function() return AiNeed(AiCityCenter()) end`